crates/sparrow-merge/src/in_memory_batches.rs

```diff
@@ -53,7 +55,9 @@ impl InMemoryBatches {
             let (version, old) = &*write;
             let version = *version;
 
-            let merged = if old.num_rows() == 0 {
+            let merged = if !self.retained {
+                old.clone()
+            } else if old.num_rows() == 0 {
                 batch.clone()
             } else {
                 homogeneous_merge(&self.schema, vec![old.clone(), batch.clone()])
```

Comment:
Note on below: The entire batch (including potentially late data) is sent to the materialization. I believe the materialization will then be able to choose how to handle that data.